dist/dialog.build.js

```diff
@@ -343,22 +395,39 @@ window.jQuery = window.jQuery || window.shoestring;
 			});
 
 			// close on escape key
-			$( doc ).bind( "keyup", function( e ){
+			$( doc ).bind( "keyup", onKeyup = function( e ){
 				if( e.which === 27 ){
 					dialog.close();
 				}
 			});
 
 			dialog._checkInteractivity();
 			var resizepoll;
-			$( window ).bind( "resize", function(){
+			$( window ).bind( "resize", onResize = function(){
 				if( resizepoll ){
 					clearTimeout( resizepoll );
 				}
 				resizepoll = setTimeout( function(){
 					dialog._checkInteractivity.call( dialog );
 				}, 150 );
-			} );
+			});
+
+			$el.bind("destroy", function(){
+				$(w).unbind("hashchange", onHashchange);
+
+				$el
+					.unbind( Dialog.events.open, onOpen )
+					.unbind( Dialog.events.close, onClose )
+					.unbind( "click", onClick );
+
+				dialog.$background.unbind( "click", onBackgroundClick);
+
+				$( doc ).unbind( "click", onDocClick );
+				$( doc ).unbind( "keyup", onKeyup );
+				$( window ).unbind( "resize", onResize );
```

Comment:
It seems like a lot of this should go in the destroy method. The event should be use the plugin prefix either way.